OpenUpSA/wazimap-ng-ui
# Fixed check for missing chart configuration

## Description

The code expected undefined but received {} instead. Moved the data validation to dataobjects so that once the data is processed by dataobjects, the rest of the codebase can assume that it is pristine which can reduce the amount of defensive programming required.

## Related Issue
No issue was raised.

## How to test it locally
Load up a profile where an indicator has a missing chart configuration, check that the default formatting is set as per default.js

## Screenshots
N/A

## Changelog

### Added
defaults.js - includes helper functions to check for missing data as well as default values
utils.test.js
defaults.test.js

### Updated
dataobjects now applies the default chart_configuration when the data is received from API is it is missing.

### Removed
removed code in chart.js that checks for a missing configuration.


## Checklist


### Pull Request


### Commits


### Code Quality


### Testing
